### PR TITLE
[flutter_lints] version 2.0

### PR DESCRIPTION
**Pull request by goderbauer**
See https://github.com/dart-lang/lints/projects/1.

Fixes https://github.com/dart-lang/lints/issues/63
Fixes https://github.com/dart-lang/lints/issues/69

Depends on https://github.com/dart-lang/lints/pull/72. Do not merge until that change is published. (Tests will also fail until then.)

### CODE REVIEW

**Comment by goderbauer** [comment]
@christopherfujino Can you grant a test exception for this? There isn't really any code to test...

**Comment by christopherfujino** [comment]
test-exempt: configuration change that will effect downstream tests.

File: packages/flutter_lints/pubspec.yaml [hunk 1]
```diff
@@ -5,8 +5,8 @@ issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+
 version: 1.0.4
 
 environment:
-  sdk: '>=2.12.0 <3.0.0'
+  sdk: '>=2.17.0-0 <3.0.0'
```
Comment (goderbauer) [inline packages/flutter_lints/pubspec.yaml:8]: Need to verify that this is the lower bound that we want.

**Review by pq — APPROVED** [review]
👍

**Comment by mit-mit** [comment]
I tried to use the new super-dot language feature from 2.17 in a widget:

That triggers the key lint:

```dart
class OtherWidget extends StatelessWidget {
  const OtherWidget({super.key});

  @override
  Widget build(BuildContext context) {
    return Container(
      
    );
  }
}
```

```
   info • Use key in widget constructors • lib/widgets.dart:6:3 • use_key_in_widget_constructors
```

Can we remove that lint?

**Comment by goderbauer** [comment]
> Can we remove that lint?

I would prefer fixing the lint since it's not a new lint added by this PR. Looks like @pq has a fix out in https://github.com/dart-lang/linter/pull/3295. Will that one make the cut for 2.17?

**Comment by pq** [comment]
> Will that one make the cut for 2.17?

It should.  As soon as I land the fix, I'll work on a new release.  (That has some other super-parameter related fixes too.)

**Comment by goderbauer** [comment]
The publishable check is failing with

```
Packages with an SDK constraint on a pre-release of the Dart SDK should themselves be published as a pre-release version. If this package needs Dart version 2.17.0-0, consider publishing the package as a pre-release instead.
  See https://dart.dev/tools/pub/publishing#publishing-prereleases For more information on pre-releases.
```

This warning is actually expected and can be ignored as described in https://dart.dev/tools/pub/publishing#publishing-previews. What's described there is exactly what we want to do here...

I don't know how to submit this, though, without turning the entire tree red because of this warning. I don't want to disable the check altogether...

@stuartmorgan Do you maybe have a suggestion?

**Comment by goderbauer** [comment]
Looks like `flutter_plugin_tools` already has a `allow-pre-release` flag. I don't think I want to turn that on for all packages. Maybe we should extend the flag to allow specifying package names for which pre-releases are allowed and then specify this one for it?

**Comment by stuartmorgan** [comment]
I think it would be fine to turn it on for the whole repo. The background is that we originally added the support unconditionally for NNBD, then when we finished that transition we kept the code around in case we needed it again, but put it behind a flag since we didn't have an actual use case in mind.

I think we can start with the easiest thing, which is turning the flag on, and then revisit whether we need to add package-level targeting if we see issues in practice.

**Comment by goderbauer** [comment]
Cool. I will make it so. Thanks!

**Pull request merged**